src/utilities/utility.ts

```diff
@@ -47,6 +50,50 @@ export function checkForErrors(execResults, warnIfError?: boolean) {
     }
 }
 
+export function annotateChildPods(kubectl: Kubectl, resourceType: string, resourceName: string, allPods): IExecSyncResult[] {
+    const commandExecutionResults = [];
+    let owner = resourceName;
+    if (resourceType.toLowerCase().indexOf('deployment') > -1) {
+        owner = kubectl.getNewReplicaSet(resourceName);
+    }
+
+    if (allPods && allPods.items && allPods.items.length > 0) {
+        allPods.items.forEach((pod) => {
+            const owners = pod.metadata.ownerReferences;
+            if (owners) {
+                owners.forEach(ownerRef => {
+                    if (ownerRef.name === owner) {
+                        commandExecutionResults.push(kubectl.annotate('pod', pod.metadata.name, workflowAnnotations, true));

```

Comment:
break after this maybe?

Reply:
refactored from `forEach` to `for..of` loop as break/return doesn't work.